### PR TITLE
fix: posthog silent fails

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: #11817 

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Review by rephus — APPROVED** [review]

File: packages/backend/src/postHog.ts [hunk 1]
```diff
@@ -47,12 +77,16 @@ export async function isFeatureFlagEnabled(
...
                 : {},
-        ),
-        new Promise<boolean>((resolve) => {
-            setTimeout(() => {
-                resolve(false);
-            }, FLAG_CHECK_TIMEOUT);
-        }),
+        );
+
+        clearTimeout(timeout);
```
Comment (almeidabbm) [inline packages/backend/src/postHog.ts:82]: This will make it so that the `timeoutPromise` never rejects or resolves

Comment (IrakliJani) [reply packages/backend/src/postHog.ts:82]: it does not matter because its in a `Promise.race`

Comment (almeidabbm) [reply packages/backend/src/postHog.ts:82]: true, what I meant is that it'll hang in the background, but probably will be garbage collected anyway and it is not important 🤔

**Review by almeidabbm — APPROVED** [review]

File: .env.development [hunk 2]
```diff
@@ -36,7 +36,7 @@ DBT_PROJECT_DIR=.
 
 SCHEDULER_ENABLED=true
 
-GROUPS_ENABLED=true
+# GROUPS_ENABLED=true
```
Comment (almeidabbm) [inline .env.development:39]: Not sure if intended, but it's for dev so not important

File: packages/backend/src/postHog.ts (hunk 1 above)
Comment (IrakliJani) [reply packages/backend/src/postHog.ts:84]: 🤞

**Review by rephus — APPROVED** [review]

**Pull request merged**